### ISSUE TITLE
Can't redirect or pipe terraform console input on Windows

### ISSUE THREAD

**Issue by ptixed** [hashicorp/terraform#29770]
### Terraform Version
```
Terraform v1.0.9
on windows_amd64
```

### Terraform Configuration Files
none

### Expected Behavior
should print 6

### Actual Behavior
command spawns terraform console without prompt symbol that cannot be interacted with nor closed

### Steps to Reproduce
```
echo "1 + 5" | terraform console
```

### Additional Context
Piping input does not work on windows 10 neither in cmd, powershell nor git bash, tested on 2 PCs. 
Running the command on linux works fine. 

### References
- #28027

**Comment by ptixed** [comment]
Hey,
Thanks for the quick reply.
I actually had the same idea yesterday after looking at the code some more. Setting `TF_FORK=0` does indeed resolve this issue for me.

**Comment by apparentlymart** [comment]
Thanks for confirming!

Given that, it does indeed seem like the "panicwrap" mechanism is causing trouble here. That's a long-standing hack we have in Terraform which has Terraform immediately start a child process and re-run itself on startup, with stderr connected to a pipe back to the parent process instead of directly to the console, and then it monitors stderr for a panic message for the Go runtime so it can handle it (marginally) more gracefully.

I originally expected an earlier change to have improved this because we reworked how Terraform sets up its terminal helper package in a way that ought to have allowed the real stdin to still be preserved, even though stderr is a pipe, but also importantly should've allowed Terraform to still detect that the _ultimate_ output is a terminal, even if it's writing out through a pipe, and thus cause `terraform console` to enter the non-terminal processing mode instead of the terminal processing mode.

This decision is ultimately made by calling the `StdinPiped` helper method:

https://github.com/hashicorp/terraform/blob/cdd5ee6fb38a6868605996f61afa74c7ba81a452/internal/command/meta.go#L346-L355

That in turn runs some platform-specific code to sense whether stdin is a terminal:

https://github.com/hashicorp/terraform/blob/cdd5ee6fb38a6868605996f61afa74c7ba81a452/internal/terminal/impl_windows.go#L83-L110

In our case here I think we actually want this to "fall out here" (as the comment says) because stdin isn't a terminal. The default behavior it's talking about is in the `IsTerminal` method, in the case where there's no overridden implementation of `isTerminal`:

https://github.com/hashicorp/terraform/blob/cdd5ee6fb38a6868605996f61afa74c7ba81a452/internal/terminal/stream.go#L75-L80

With all of this said, I think what we need to understand here is why the "panicwrap" mechanism is making stdin look like a terminal even though it doesn't look like a terminal in normal mode. But our panicwrap fixup behavior only does anything special for stderr, so I'm not sure why it's affecting the stdin behavior:

https://github.com/hashicorp/terraform/blob/cdd5ee6fb38a6868605996f61afa74c7ba81a452/internal/terminal/panicwrap_ugh.go#L53-L71

I'd guess that the problem is somewhere in the vicinity of here:

https://github.com/hashicorp/terraform/blob/cdd5ee6fb38a6868605996f61afa74c7ba81a452/internal/helper/wrappedstreams/streams_windows.go#L12-L53

Specifically, it seems like opening `CONIN$` here is ending up holding a filehandle for the console, rather than a filehandle for the input pipe, and so we ultimately end up detecting this as being a terminal.

I've run out of time to research this further today, but I just wanted to leave some breadcrumbs in case it's helpful to someone else picking this up later.

---

I think while we're looking at this it might also be good to think about whether benefits of the panicwrap hack are greater than its apparently-high maintenance cost, since we've repeatedly run into various quirks caused by this unusual step of self-forking and writing through pipes.

As far as I'm aware, the benefits it provides are:

* It automatically captures Terraform's internal log output, which is normally hidden, into a `crash.log` file if it detects an unhandled panic. That therefore makes it a little easier to file a bug report, because you already have the trace log that led to that particular panic and therefore don't need to figure out how to reproduce it with the `TF_LOG=trace` environment variable enabled.
* It prints out a special message prompting users to open a bug report:

    https://github.com/hashicorp/terraform/blob/cdd5ee6fb38a6868605996f61afa74c7ba81a452/internal/logging/panic.go#L18-L35

It seems then that the main cost of removing it would be that users would need to take extra steps in order to create a trace log file to open a bug report, which would make opening a bug report about an unhandled panic require roughly the same effort as opening a bug report about any other (non-panic) unexpected error, following the instructions included in our issue template:

https://github.com/hashicorp/terraform/blob/cdd5ee6fb38a6868605996f61afa74c7ba81a452/.github/ISSUE_TEMPLATE/bug_report.md#L49-L54

I will be up front about my bias towards removing it, since we've now invested very many hours in trying to work around its quirks at a high opportunity cost, but I do think it would be worth at least trying to root-cause this particular issue on Windows first, and then we can make an informed decision based on how much effort we think it'll take to fix it.

**Comment by jbardin** [comment]
Closed via #29825
